src/runtime/mod.rs

```diff
@@ -39,6 +39,42 @@ pub const DEFAULT_BINDLE_SERVER: &str = "http://localhost:8080/v1";
 const WASM_LAYER_CONTENT_TYPE: &str = "application/vnd.wasm.content.layer.v1+wasm";
 const STDERR_FILE: &str = "module.stderr";
 
+pub struct RouterInfo {
+    pub entrypoint: String,
+    pub client_addr: SocketAddr,
+    pub cache_config_path: PathBuf,
+    pub module_cache_dir: PathBuf,
+    pub base_log_dir: PathBuf,
+    pub default_host: String,
+    pub use_tls: bool,
+    pub env_vars: HashMap<String, String>,
+}
+
+impl RouterInfo {
```

Comment:
This is a nit, but usually in Rust, you'd do the same thing here that you would do in Go: Instead of having a `new()` with params for every field, you'd just declare the struct inline. That said, there's no strong idiom that I know of that says you _can't_ have a `new()` like this. So I suppose that at the end of the day it's just a matter of taste. Any thoughts, @thomastaylor312 ?

Reply:
Yea that makes sense, especially since there are no other associated methods, a constructor would be superfluous.

Reply:
done